### PR TITLE
Add target for Eigen linear algebra libraries

### PR DESCRIPTION
**Pull request by dodsonmg**
[Eigen's website](http://eigen.tuxfamily.org/index.php?title=Main_Page)

### CODE REVIEW

File: pycheribuild/projects/cross/eigen.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# from ..project import ReuseOtherProjectRepository
+
+
+class BuildEigen(CrossCompileCMakeProject):
+    project_name = "eigen"
+    repository = GitRepository("https://gitlab.com/libeigen/eigen.git")
+    native_install_dir = DefaultInstallDir.IN_BUILD_DIRECTORY
+    cross_install_dir = DefaultInstallDir.SYSROOT
+
+    @classmethod
```
Comment (arichardson) [inline pycheribuild/projects/cross/eigen.py:41]: None of these methods need to be defined if they just call the superclass.

File: pycheribuild/projects/cross/eigen.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#
+# Copyright (c) 2020 Alex Richardson
```
Comment (arichardson) [inline pycheribuild/projects/cross/eigen.py:2]: You probably want your copyright here instead of listing me.

Comment (dodsonmg) [reply pycheribuild/projects/cross/eigen.py:2]: Well, you wrote cheribuild and the [whatever].py script on which eigen.py was based.  I feel like my modification of the script is similar to providing a command line argument.

File: pycheribuild/projects/cross/eigen.py (hunk 1 above)
Comment (dodsonmg) [reply pycheribuild/projects/cross/eigen.py]: removed.

**Pull request merged**